qmi/instruments/newport/single_axis_motion_controller.py

```diff
@@ -622,22 +641,23 @@ def get_acceleration(self, controller_address: Optional[int] = None) -> float:
         Returns:
             Encoder increment value.
         """
+        def _get_acceleration() -> str:
+            acceleration = self._scpi_protocol.ask(
+                self._build_command("AC?"))
+            sleep(self.COMMAND_EXEC_TIME)
+            self._check_error()
+            return acceleration
+
         _logger.info("Getting acceleration of instrument [%s]", self._name)
         self.controller_address = controller_address
         # instrument must be in CONFIGURATION, DISABLE or READY state to get the acceleration.
         try:
```

Comment:
I see that in a lot of commands we do this `try` `catch` to check if the instrument is already in the ready state. Perhaps we can do this in the context manager. Then we can validate the ready state in the context manager instead of repeating the check for each method that needs it. The commands would like something like:
```python
with self.configuration_state("acceleration"):
    acceleration = _get_acceleration()
```

Reply:
also then you don't need a private method like `_get_acceleration()`, because I think you made it so you could use the same in the `try` and `catch` block.

Reply:
I think this would not work. This is because we use the `try`-`except` block to decide if we need to use the context manager. Moving the check inside the context manager defeats its purpose as we will then always be using the context manager even when it is not needed.